pkg/console/operator/sync_v400.go

```diff
@@ -114,10 +114,16 @@ func sync_v400(co *consoleOperator, consoleConfig *v1alpha1.Console) (*v1alpha1.
 func SyncDeployment(co *consoleOperator, recorder events.Recorder, consoleConfig *v1alpha1.Console, cm *corev1.ConfigMap, serviceCAConfigMap *corev1.ConfigMap, sec *corev1.Secret) (*appsv1.Deployment, bool, error) {
 	logrus.Printf("validating console deployment...")
 	defaultDeployment := deploymentsub.DefaultDeployment(consoleConfig, cm, serviceCAConfigMap, sec)
-	versionAvailability := &operatorv1alpha1.VersionAvailability{
-		Version: consoleConfig.Spec.Version,
+	versionAvailability := []operatorv1.GenerationStatus{
+		{
+			Group:          "apps/v1",
+			Resource:       "deployments",
+			Namespace:      defaultDeployment.Namespace,
+			Name:           defaultDeployment.Name,
+			LastGeneration: defaultDeployment.Generation,
+		},
 	}
-	deploymentGeneration := resourcemerge.ExpectedDeploymentGenerationV1alpha1(defaultDeployment, versionAvailability)
+	deploymentGeneration := resourcemerge.ExpectedDeploymentGeneration(defaultDeployment, versionAvailability)
```

Comment:
@deads2k can you please confirm that this is the right usage of this function? cause I haven't found any other usage across other operator repositories. Thanks !

Reply:
Yeah, I found this one a touch tricky as well.